packages/components/managed-data-table/src/data-table-settings/column-manager/column-manager.js

```diff
@@ -0,0 +1,231 @@
+                      onDragStart={handleDragStart}
+                    >
+                      <Spacings.Inline scale="m">
+                        <DroppableContainer isDragging={isDragging}>
+                          <Spacings.Stack>
+                            <Spacings.Inline scale="xs" alignItems="center">
+                              <EyeCrossedIcon size="medium" />
+                              <FieldLabel
+                                title={formatMessage(messages.hiddenColumns)}
+                              />
+                            </Spacings.Inline>
+                            {props.areHiddenColumnsSearchable && (
+                              <AsyncSelectInput
+                                id="hidden-columns-panel-search-input"
+                                {...(props.searchHiddenColumnsPlaceholder
+                                  ? {
+                                      placeholder:
+                                        props.searchHiddenColumnsPlaceholder,
+                                    }
+                                  : undefined)}
+                                cacheOptions={false}
+                                onChange={() => {
+                                  // to avoid prop-types error
+                                }}
+                                // loadOptions is used instead of onInputChange
+                                // because the loading indicator is displayed
+                                // only when loadOptions is invoked
+                                loadOptions={debounce(
+                                  props.searchHiddenColumns,
+                                  300
+                                )}
+                                onInputChange={(input) =>
+                                  // loadOptions is not invoked when input is empty
+                                  !input.length &&
+                                  props.searchHiddenColumns(input)
+                                }
+                                components={{
+                                  Option: () => null,
+                                  Menu: () => null,
+                                  // eslint-disable-next-line react/display-name
+                                  DropdownIndicator: () => (
+                                    <Spacings.Inline alignItems="center">
+                                      <SearchIcon
+                                        scale="medium"
+                                        color="primary"
+                                      />
+                                    </Spacings.Inline>
+                                  ),
+                                }}
+                              />
+                            )}
+                            <DroppablePanel
+                              droppableId="hidden-columns-panel"
+                              data-testid="hidden-columns-panel"
```

Comment:
Should we put this into a constant on top of this file the least? Feels like it's reused quite a bit.